src/vellum/workflows/resolvers/base.py

```diff
@@ -1,21 +1,26 @@
 from abc import ABC, abstractmethod
 from uuid import UUID
-from typing import TYPE_CHECKING, Iterator, Optional
+from typing import TYPE_CHECKING, Iterator, Optional, Type
 
 from vellum.workflows.events.workflow import WorkflowEvent
 from vellum.workflows.state.base import BaseState
 
 if TYPE_CHECKING:
     from vellum.workflows.state.context import WorkflowContext
+    from vellum.workflows.workflows.base import BaseWorkflow
 
 
 class BaseWorkflowResolver(ABC):
     def __init__(self):
         self._context: Optional["WorkflowContext"] = None
+        self._workflow_class: Optional[Type["BaseWorkflow"]] = None
 
     def register_context(self, context: "WorkflowContext") -> None:
         self._context = context
 
+    def register_workflow(self, workflow_class: Type["BaseWorkflow"]) -> None:
```

Comment:
[nit] if we pass in the instance here, we could save both the class and the context with one register method